onnx/checker.py

```diff
@@ -90,7 +95,12 @@ def check_model(model, full_check=False):  # type: (Union[ModelProto, Text], boo
         C.check_model_path(model)
         m = onnx.load(model)
     else:
-        C.check_model(model.SerializeToString())
+        # If the protobuf is larger than 2GB,
+        # remind user should use the model path to check
+        protobuf_string = model.SerializeToString()
+        if sys.getsizeof(protobuf_string) > MAXIMUM_PROTOBUF:
+            warnings.warn(str('This protobuf of onnx model is too large (>2GB). Call check_model with model path instead.'))
```

Comment:
Does warn throw an exception? Wouldn't it be better for the check_model to be in the else branch?

Reply:
Warn just throws a message without interrupting the code. 
Sounds good. Models larger than 2gb won't pass the checker anyway so it is no need to run the checker.